lib/tpm2_auth_util.c

```diff
@@ -48,8 +50,88 @@ static bool handle_hex_password(const char *password, TPM2B_AUTH *auth) {
     return true;
 }
 
+static bool handle_password(const char *password, TPM2B_AUTH *auth);
+
+static tool_rc get_auth_for_file_param(const char* password, TPM2B_AUTH *auth) {
```

Comment:
Would `tpm2_auth_util_from_optarg()` be better here to avoid code duplication?

Reply:
`tpm2_auth_util_from_optarg` also allows the prefix `session` or `pcr` which would not be appropriate.
There is not so much code duplication `get_auth_for_file_param` uses and fixes code from `handle_file` which now calls `get_auth_for_file_param`.

Reply:
Another option would be to make an _ex variant that takes flags for what it could support, but I'll leave that judgment call on you. Both roads lead to Rome.

Reply:
@williamcroberts next week i'll think about what's the best way to rome ;-)

Reply:
@williamcroberts I would prefer the current solution, because there is a clear separation
in the way of the password handling between `tpm2_auth_util_from_optarg` and  `handle_str_password`.
So there is no need for addtional flags...

Reply:
I would prefer a `tpm2_auth_util_from_optarg_ex()` that takes a flag like @williamcroberts proposes.  The reasons for this are: a) the delta to `tpm2_auth_util_from_optarg()` will be smaller than the delta here, b) you might need the same functionality again later, and doing that now saves you or someone else the refactor later.

Reply:
I would say there is no big delta. Some code was moved to the new function `get_auth_for_file_param` with
the fixes you asked for and the fix for reallocation in getline. And I would say it'ts better do use the function
`handle_str_password` to get the auth structure than using a function which returns a session which is used to get the auth structure.